.gitignore

```diff
@@ -1,4 +1,5 @@
 /vendor
+/.vscode
```

Comment:
There are a lot of personal preferences there. I'd rather not have somebody else's settings. Probably exclude the settings file only, then (`.vscode/settings.json`).

Reply:
I'm fine with adding this here, it's not very intrusive and `.vscode` would probably be the biggest use case for the language server.

Reply:
yes, i just meant to imply that the more of a userbase we have the more contributors we are likely to have

Reply:
I'm not sure to see the problem with using `.git/info/exclude` but it's not a big deal, @BlademMF you can resolve (I can't)

Reply:
@camilledejoye, if I am reading this right, this setting will be local for me. The problem with that approach is if another guy who uses VS Code contributes and then his settings end up in the repo which defeats the purpose, because they will interfere with mine.

Reply:
We wouldn't let anybody commit their `.vscode` folder though :)

Reply:
I that case, I'll do whatever you say. But isn't that the purpose of `.gitignore` - to not let people commit stuff?

Reply:
For me the goal is to avoid me the pain to deal with it but I get your point.
I personally always use my global gitignore when it comes to stuff that will impact multiple project and the `.git` directory for stuff that are specific to a project.

Reply:
Yes, I've added this to all Phpactor repos already, but yes, this file should be used for project-specific things (and we shouldn't add the ignore patterns for every single IDE ever made)

Reply:
I totally believe you have good reasons for doing so, but for me the `.gitignore` file is sort of "The rules of the repository" and it goes with the repository for everyone to follow. I could be missing something and I would be grateful if you explained if I did.

Reply:
> and we shouldn't add the ignore patterns for every single IDE ever made

I get your point - people should be well-mannered enough to not commit their stuff. I will do whatever you instruct me to.

Reply:
What should I do here?

Reply:
well, I've added it to all the repos (or most of them, not sure it worked properly), so leave it. But as a policy I think we should not have IDE specific patterns here.